src/Hypernode/Magento2/Command/Hypernode/Crack/AbstractCrackCommand.php

```diff
@@ -0,0 +1,469 @@
+                'user'     => $credential->getId(),
+                'hash'     => $credential->getHash(),
+                'cracked'  => $credential->isCracked() ? '1' : '0',
+                'password' => $credential->getPassword(),
+            );
+        }
+        $headings = array('User', 'Hash', 'Cracked', 'Password');
+        $t = $this->getHelper('table');
+        $t->setHeaders($headings)->renderByFormat($output, $results, 'csv');
+    }
+
+    protected function getWordFiles()
+    {
+        $wordlists = $this->input
+            ->getArgument('wordlists');
+
+        $special = array_search('special', $wordlists);
+        if ($special !== false) {
+            $filePath = $this->generateSpecialWordlist();
+            $wordlists = array_replace($wordlists, array($special => $filePath));
+        }
+
+        $resolver = new FileResolver($wordlists, $this->config['wordlistDirs'], 'txt');
+        $invalid = $resolver->getInvalidFiles();
+        if (!empty($invalid)) {
+            throw new \InvalidArgumentException(
+                sprintf('Wordlist file(s) not found [%s].', implode(', ', $invalid))
+            );
+        }
+
+        return $resolver->getValidFiles();
+    }
+
+    protected function getRuleFiles()
+    {
+        $usedRuleSets = $this->input->getOption('rulesets');
+
+        $resolver = new FileResolver($usedRuleSets, $this->config['rulesDirs'], 'rule');
+
+        $invalid = $resolver->getInvalidFiles();
+        if (!empty($invalid)) {
+            throw new \InvalidArgumentException(
+                sprintf('Ruleset file(s) not found [%s].', implode(', ', $invalid))
+            );
+        }
+
+        return $resolver->getValidFiles();
+    }
+
+    protected function filterCracked($results, $cracked)
+    {
+        $filteredResults = array();
+        foreach ($results as $result) {
+            if ($result->isCracked() === $cracked) {
+                $filteredResults[] = $result;
+            }
+        }
+
+        return $filteredResults;
+    }
+
+    protected function filterOutput($results)
+    {
+        $input     = $this->input;
+        $cracked   = $input->getOption('cracked');
+        $uncracked = $input->getOption('uncracked');
+        if ($cracked && ! $uncracked) {
+            return $this->filterCracked($results, true);
+        } elseif ($uncracked && ! $cracked) {
+            return $this->filterCracked($results, false);
+        }
+
+        return $results;
+    }
+
+    protected function generateSpecialWordlist()
+    {
+        $admins = $this->userModel->getCollection();
+        $stores = $this->storeManager->getStores();
+        $generator = new WordlistGenerator();
+        $generator->setAdmins($admins);
+        $generator->setStores($stores);
+        $words = $generator->generate();
+
+        $directoryRead = $this->filesystem->getDirectoryRead('tmp');
+        $dir = $directoryRead->getAbsolutePath('password-cracker');
+
+        $io = new \Varien_Io_File;
```

Comment:
I don't think this class exists in M2, should be replaced with \Magento\Framework\Filesystem\Io\File;